homeassistant/components/dremel_3d_printer/__init__.py

```diff
@@ -0,0 +1,155 @@
+"""The Dremel 3D Printer (3D20, 3D40, 3D45) integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from dremel3dpy import Dremel3DPrinter
+from requests.exceptions import ConnectTimeout, HTTPError
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+import homeassistant.util.dt as dt_util
+
+from .const import _LOGGER, DOMAIN, PLATFORMS, SCAN_INTERVAL
+from .services import async_setup_services
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the OctoPrint component."""
+    if DOMAIN not in config:
+        return True
+
+    domain_config = config[DOMAIN]
+
+    for conf in domain_config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data={
+                    CONF_HOST: conf[CONF_HOST],
+                },
+            )
+        )
+
+    return True
```

Comment:
This looks out of place

Reply:
If I search for "async def async_setup(" all the usages I see are defined in __init__.py. I was inspired by the integration `octoprint`. Please let me know how you'd go with placing it correctly. Thanks!